[
    {
        "timestamp": 1.41131591796875,
        "seqNumber": 64,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal Yaw",
        "callStack": "com.ctre.phoenix6.jni.ErrorReportingJNI.reportStatusCode(Native Method)\ncom.ctre.phoenix6.StatusSignal.updateValue(StatusSignal.java:210)\ncom.ctre.phoenix6.StatusSignal.refresh(StatusSignal.java:230)\ncom.ctre.phoenix6.hardware.ParentDevice.commonLookup(ParentDevice.java:323)\ncom.ctre.phoenix6.hardware.ParentDevice.lookupStatusSignal(ParentDevice.java:333)\ncom.ctre.phoenix6.hardware.core.CorePigeon2.getYaw(CorePigeon2.java:324)\ncom.ctre.phoenix6.hardware.Pigeon2.<init>(Pigeon2.java:41)\ncom.ctre.phoenix6.hardware.Pigeon2.<init>(Pigeon2.java:78)\nfrc.robot.subsystems.Drivetrain.<init>(Drivetrain.java:116)\nfrc.robot.RobotContainer.<init>(RobotContainer.java:13)\nfrc.robot.Robot.robotInit(Robot.java:18)\nedu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:107)\nedu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)\nedu.wpi.first.wpilibj.RobotBase.startRobot(RobotBase.java:453)\nfrc.robot.Main.main(Main.java:13)\n",
        "messageType": 0
    },
    {
        "timestamp": 1.41265869140625,
        "seqNumber": 65,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal AngularVelocityZWorld",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "timestamp": 1.41339111328125,
        "seqNumber": 66,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal QuatW",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "timestamp": 1.4140625,
        "seqNumber": 67,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal QuatX",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "timestamp": 1.41473388671875,
        "seqNumber": 68,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal QuatY",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "timestamp": 1.4154052734375,
        "seqNumber": 69,
        "numOccur": 1,
        "errorCode": -1003,
        "flags": 1,
        "details": "CAN frame not received/too-stale. Check the CAN bus wiring, CAN bus utilization, and power to the device.",
        "location": "pigeon 2 0 () Status Signal QuatZ",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "timestamp": 1.576416015625,
        "seqNumber": 70,
        "numOccur": 1,
        "errorCode": -1001,
        "flags": 1,
        "details": "Could not transmit CAN Frame.",
        "location": "pigeon 2 0 () Apply Config",
        "callStack": "Stack Trace not received for performance reasons.",
        "messageType": 0
    },
    {
        "messageType": 2,
        "timestamp": 1.80584716796875,
        "seqNumber": 71,
        "line": "********** Robot program startup complete **********"
    },
    {
        "timestamp": 1.9071044921875,
        "seqNumber": 72,
        "numOccur": 1,
        "errorCode": 1,
        "flags": 0,
        "details": "Loop time of 0.02s overrun\n",
        "location": "edu.wpi.first.wpilibj.IterativeRobotBase.printLoopOverrunMessage(IterativeRobotBase.java:412)",
        "callStack": "",
        "messageType": 1
    },
    {
        "messageType": 2,
        "timestamp": 1.90777587890625,
        "seqNumber": 73,
        "line": "Warning at edu.wpi.first.wpilibj.IterativeRobotBase.printLoopOverrunMessage(IterativeRobotBase.java:412): Loop time of 0.02s overrun"
    },
    {
        "messageType": 2,
        "timestamp": 1.90875244140625,
        "seqNumber": 74,
        "line": ""
    },
    {
        "timestamp": 1.91741943359375,
        "seqNumber": 75,
        "numOccur": 1,
        "errorCode": 1,
        "flags": 1,
        "details": "Unhandled exception: java.lang.NullPointerException: Cannot invoke \"edu.wpi.first.math.kinematics.SwerveDriveOdometry.update(edu.wpi.first.math.geometry.Rotation2d, edu.wpi.first.math.kinematics.SwerveModulePosition[])\" because \"this.simOdometry\" is null",
        "location": "frc.robot.subsystems.Drivetrain.updatePoseEstimator(Drivetrain.java:261)",
        "callStack": "\tat frc.robot.subsystems.Drivetrain.updatePoseEstimator(Drivetrain.java:261)\n\tat frc.robot.subsystems.Drivetrain.periodic(Drivetrain.java:187)\n\tat edu.wpi.first.wpilibj2.command.CommandScheduler.run(CommandScheduler.java:260)\n\tat frc.robot.Robot.robotPeriodic(Robot.java:23)\n\tat edu.wpi.first.wpilibj.IterativeRobotBase.loopFunc(IterativeRobotBase.java:381)\n\tat edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:131)\n\tat edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)\n\tat edu.wpi.first.wpilibj.RobotBase.startRobot(RobotBase.java:453)\n\tat frc.robot.Main.main(Main.java:13)\n",
        "messageType": 0
    },
    {
        "messageType": 2,
        "timestamp": 1.918212890625,
        "seqNumber": 76,
        "line": "Error at frc.robot.subsystems.Drivetrain.updatePoseEstimator(Drivetrain.java:261): Unhandled exception: java.lang.NullPointerException: Cannot invoke \"edu.wpi.first.math.kinematics.SwerveDriveOdometry.update(edu.wpi.first.math.geometry.Rotation2d, edu.wpi.first.math.kinematics.SwerveModulePosition[])\" because \"this.simOdometry\" is null"
    },
    {
        "timestamp": 1.9183349609375,
        "seqNumber": 77,
        "numOccur": 1,
        "errorCode": 1,
        "flags": 0,
        "details": "The robot program quit unexpectedly. This is usually due to a code error.\n  The above stacktrace can help determine where the error occurred.\n  See https://wpilib.org/stacktrace for more information.",
        "location": "edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:379)",
        "callStack": "",
        "messageType": 1
    },
    {
        "messageType": 2,
        "timestamp": 1.91864013671875,
        "seqNumber": 78,
        "line": "\tat frc.robot.subsystems.Drivetrain.updatePoseEstimator(Drivetrain.java:261)"
    },
    {
        "messageType": 2,
        "timestamp": 1.919189453125,
        "seqNumber": 80,
        "line": "\tat frc.robot.subsystems.Drivetrain.periodic(Drivetrain.java:187)"
    },
    {
        "timestamp": 1.91912841796875,
        "seqNumber": 79,
        "numOccur": 1,
        "errorCode": 1,
        "flags": 1,
        "details": "The startCompetition() method (or methods called by it) should have handled the exception above.",
        "location": "edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:386)",
        "callStack": "",
        "messageType": 0
    },
    {
        "messageType": 2,
        "timestamp": 1.91961669921875,
        "seqNumber": 81,
        "line": "\tat edu.wpi.first.wpilibj2.command.CommandScheduler.run(CommandScheduler.java:260)"
    },
    {
        "messageType": 2,
        "timestamp": 1.919921875,
        "seqNumber": 82,
        "line": "\tat frc.robot.Robot.robotPeriodic(Robot.java:23)"
    },
    {
        "messageType": 2,
        "timestamp": 1.92034912109375,
        "seqNumber": 83,
        "line": "\tat edu.wpi.first.wpilibj.IterativeRobotBase.loopFunc(IterativeRobotBase.java:381)"
    },
    {
        "messageType": 2,
        "timestamp": 1.920654296875,
        "seqNumber": 84,
        "line": "\tat edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:131)"
    },
    {
        "messageType": 2,
        "timestamp": 1.92095947265625,
        "seqNumber": 85,
        "line": "\tat edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)"
    },
    {
        "messageType": 2,
        "timestamp": 1.92169189453125,
        "seqNumber": 86,
        "line": "\tat edu.wpi.first.wpilibj.RobotBase.startRobot(RobotBase.java:453)"
    },
    {
        "messageType": 2,
        "timestamp": 1.92205810546875,
        "seqNumber": 87,
        "line": "\tat frc.robot.Main.main(Main.java:13)"
    },
    {
        "messageType": 2,
        "timestamp": 1.9224853515625,
        "seqNumber": 88,
        "line": ""
    },
    {
        "messageType": 2,
        "timestamp": 1.92333984375,
        "seqNumber": 89,
        "line": "Warning at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:379): The robot program quit unexpectedly. This is usually due to a code error."
    },
    {
        "messageType": 2,
        "timestamp": 1.923583984375,
        "seqNumber": 90,
        "line": "  The above stacktrace can help determine where the error occurred."
    },
    {
        "messageType": 2,
        "timestamp": 1.92388916015625,
        "seqNumber": 91,
        "line": "  See https://wpilib.org/stacktrace for more information."
    },
    {
        "messageType": 2,
        "timestamp": 1.9241943359375,
        "seqNumber": 92,
        "line": "Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:386): The startCompetition() method (or methods called by it) should have handled the exception above."
    },
    {
        "timestamp": 1.92901611328125,
        "seqNumber": 93,
        "numOccur": 1,
        "errorCode": 0,
        "flags": 0,
        "details": "[phoenix-diagnostics] Server shutdown cleanly. (dur:0)",
        "location": "",
        "callStack": "",
        "messageType": 1
    },
    {
        "timestamp": 1.92950439453125,
        "seqNumber": 94,
        "numOccur": 1,
        "errorCode": 0,
        "flags": 0,
        "details": "[phoenix] Library shutdown cleanly",
        "location": "",
        "callStack": "",
        "messageType": 1
    },
    {
        "messageType": 2,
        "timestamp": 1.95428466796875,
        "seqNumber": 95,
        "line": "#"
    },
    {
        "messageType": 2,
        "timestamp": 1.954833984375,
        "seqNumber": 96,
        "line": "# A fatal error has been detected by the Java Runtime Environment:"
    },
    {
        "messageType": 2,
        "timestamp": 1.955078125,
        "seqNumber": 97,
        "line": "#"
    },
    {
        "messageType": 2,
        "timestamp": 1.955322265625,
        "seqNumber": 98,
        "line": "#  SIGSEGV (0xb) at pc=0xb6dfffc0, pid=3077, tid=3081"
    },
    {
        "messageType": 2,
        "timestamp": 1.95550537109375,
        "seqNumber": 99,
        "line": "#"
    },
    {
        "messageType": 2,
        "timestamp": 1.95587158203125,
        "seqNumber": 100,
        "line": "# JRE version: OpenJDK Runtime Environment (17.0.9.7) (build 17.0.9.7-frc+0-2024-17.0.9u7-1)"
    },
    {
        "messageType": 2,
        "timestamp": 1.9561767578125,
        "seqNumber": 101,
        "line": "# Java VM: OpenJDK Client VM (17.0.9.7-frc+0-2024-17.0.9u7-1, mixed mode, emulated-client, serial gc, linux-arm)"
    },
    {
        "messageType": 2,
        "timestamp": 1.95654296875,
        "seqNumber": 102,
        "line": "# Problematic frame:"
    },
    {
        "messageType": 2,
        "timestamp": 1.9580078125,
        "seqNumber": 103,
        "line": "# C  [libc.so.6+0x6bfc0]"
    },
    {
        "messageType": 2,
        "timestamp": 1.96148681640625,
        "seqNumber": 104,
        "line": "#"
    },
    {
        "messageType": 2,
        "timestamp": 1.9620361328125,
        "seqNumber": 105,
        "line": "# No core dump will be written. Core dumps have been disabled. To enable core dumping, try \"ulimit -c unlimited\" before starting Java again"
    },
    {
        "messageType": 2,
        "timestamp": 1.9622802734375,
        "seqNumber": 106,
        "line": "#"
    },
    {
        "messageType": 2,
        "timestamp": 1.96282958984375,
        "seqNumber": 107,
        "line": "# An error report file with more information is saved as:"
    },
    {
        "messageType": 2,
        "timestamp": 1.96319580078125,
        "seqNumber": 108,
        "line": "# /tmp/hs_err_pid3077.log"
    }
]